scripts/get_ocp_addon_versions.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+import http.client
+import json
+import sys
+
+
+def parse_input():
+    """
+    Reads JSON input from stdin and parses it into a dictionary.
+    Returns:
+        dict: Parsed input data.
+    """
+    try:
+        data = json.loads(sys.stdin.read())
+    except json.JSONDecodeError as e:
+        raise ValueError("Invalid JSON input") from e
+    return data
+
+
+def validate_inputs(data):
+    """
+    Validates required inputs 'IAM_TOKEN' and 'REGION' from the parsed input.
+    Args:
+        data (dict): Input data parsed from JSON.
+    Returns:
+        tuple: A tuple containing (IAM_TOKEN, REGION).
+    """
+    token = data.get("IAM_TOKEN")
+    if not token:
+        raise ValueError("IAM_TOKEN is required")
+
+    region = data.get("REGION")
+    if not region:
+        raise ValueError("REGION is required")
+
+    return token, region
+
+
+def fetch_addon_versions(iam_token, region):
+    """
+    Fetches openshift add-on versions using HTTP connection.
+    Args:
+        iam_token (str): IBM Cloud IAM token for authentication.
+        region (str): Region to query for add-ons.
+    Returns:
+        list: Parsed JSON response containing add-on information.
+    """
+    url = "/global/v1/addons"
+    host = "containers.cloud.ibm.com"
```

Comment:
We need an environment variable override here to support other environments such as test.cloud.ibm.com. So you should refactor this to allow the user to pass an enviornmeent variable named `IBMCLOUD_CS_API_ENDPOINT`. If user does not specify the enviornment variable, the default value should be `https://containers.test.cloud.ibm.com/global` as per the provider docs:

<img width="798" height="51" alt="Image" src="https://github.com/user-attachments/assets/e0f21b5a-fc23-49c8-a7de-5457a8ea7a85" />